src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/register_app_interface_request.cc

```diff
@@ -340,21 +340,22 @@ void RegisterAppInterfaceRequest::Run() {
         msg_params.getElement(strings::app_hmi_type);
 
     for (size_t i = 0; i < app_type.length(); ++i) {
-      if (mobile_apis::AppHMIType::NAVIGATION ==
+      mobile_apis::AppHMIType::eType current_app_type =
           static_cast<mobile_apis::AppHMIType::eType>(
-              app_type.getElement(i).asUInt())) {
+              app_type.getElement(i).asUInt());
+
+      if (mobile_apis::AppHMIType::NAVIGATION == current_app_type) {
         application->set_is_navi(true);
       }
-      if (mobile_apis::AppHMIType::COMMUNICATION ==
-          static_cast<mobile_apis::AppHMIType::eType>(
-              app_type.getElement(i).asUInt())) {
+      if (mobile_apis::AppHMIType::COMMUNICATION == current_app_type) {
         application->set_voice_communication_supported(true);
       }
-      if (mobile_apis::AppHMIType::PROJECTION ==
-          static_cast<mobile_apis::AppHMIType::eType>(
-              app_type.getElement(i).asUInt())) {
+      if (mobile_apis::AppHMIType::PROJECTION == current_app_type) {
         application->set_mobile_projection_enabled(true);
       }
+      if (mobile_apis::AppHMIType::REMOTE_CONTROL == current_app_type) {
```

Comment:
@BSolonenko propose to replace all these if's with `switch (current_app_type)`

Reply:
@AKalinich-Luxoft  this enum contain value not handled in switch.

Reply:
@BSolonenko so what is the problem? You can handle all needed HMI type values and add `default` case for all other values with an empty implementation

Reply:
@AKalinich-Luxoft Fixed in https://github.com/smartdevicelink/sdl_core/pull/2048/commits/3ffd281451b051e9ead29c3a877a6678109321ac